lib/decimal.dart

```diff
@@ -23,6 +23,8 @@ class Decimal implements Comparable<Decimal> {
 
   factory Decimal.fromInt(int value) => Decimal.fromBigInt(BigInt.from(value));
 
+  factory Decimal.fromJson(Object? value) => Decimal.parse(value.toString());
```

Comment:
I wonder if the type shouldn't be `String`. WDYT?

Reply:
I think strings converted from numbers are valid for parse at this point, and we can't use other constructors here. Any idea about this?

Reply:
The behavior is the same as parse when you deal with JSON fields.

Reply:
One advantage of having Object is that fromJson will work with json containing number like `1`, `1.0` or strings like `"1.0"`.
On the other hand it's a little _strange_ to have different types for toJson and fromJson.

Reply:
Indeed. Using `Object?` makes every formal value pass the compile check, and throws exceptions from `parse`.
```dart
return Something(
  value: Decimal.fromJson(json['value']) // This doesn't require further convert methods, especially with `always_specify_types`.
);
```

Now I'm thinking the factory constructor cannot be used as a static method with annotations like `@JsonKey(fromJson: Decimal.fromJson)`, so it might be better if we have a static method. Sounds good to you?

Reply:
> Now I'm thinking the factory constructor cannot be used as a static method with annotations like `@JsonKey(fromJson: Decimal.fromJson)`, so it might be better if we have a static method. Sounds good to you?

If we add a `Decimal.fromJson` constructor why should we have to provide a `@JsonKey(fromJson: ...)` ? Decoding will be supported out of the box. Are you using [json_serializable](https://pub.dev/packages/json_serializable) ?

Reply:
> Are you using json_serializable ?

Yes. That might provide some conveniences when using it.